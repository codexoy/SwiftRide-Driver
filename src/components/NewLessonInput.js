import React from "react"
import { StyleSheet } from "react-native"
import { Input, Button, Icon } from "react-native-elements"
import { strings } from "../i18n"

export default class NewLessonInput extends React.Component {
	render() {
		const { name } = this.props
		let iconType = "material"
		if (this.props.iconType) iconType = this.props.iconType
		return (
			<Input
				placeholder={
					strings(`teacher.new_lesson.${name}`) +
					this.props.extraPlaceholder
				}
				onChangeText={value => this.props.onChangeText(name, value)}
				value={this.props.state[name]}
				testID={`${name}Input`}
				inputContainerStyle={{
					...styles.inputContainer,
					...this.props.style
				}}
				inputStyle={{
					...styles.input,
					...{
						color: this.props.state[`${name}Color`] || "#000"
					}
				}}
				textAlign={"right"}
				autoFocus={this.props.autoFocus || false}
				ref={input => {
					this.props.setRef(input, name)
				}}
				onSubmitEditing={() => {
					if (this.props.onSubmitEditing) {
						this.props.onSubmitEditing()
						return
					}
					if (!this.props.next) return
					this.props.next().focus()
				}}
				leftIcon={
					<Icon
						name={this.props.iconName}
						type={iconType}
						size={24}
						color={this.props.state[`${name}Color`] || "#000"}
					/>
				}
				placeholderTextColor={
					this.props.state[`${name}Color`] || "lightgray"
				}
				onFocus={() => this.props.onFocus(name)}
				onBlur={() => this.props.onBlur(name)}
				editable={this.props.editable}
				selectTextOnFocus={this.props.selectTextOnFocus}
			/>
		)
	}
}

const styles = StyleSheet.create({
	inputContainer: {
		borderBottomColor: "rgb(200,200,200)",
		borderBottomWidth: 1,
		paddingBottom: 8,
		marginTop: 24
	},
	input: {
		paddingLeft: 12
	}
})
