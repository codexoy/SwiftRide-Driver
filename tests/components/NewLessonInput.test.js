import { Text } from "react-native"
import React from "react"

import renderer from "react-test-renderer"

import NewLessonInput from "../../src/components/NewLessonInput"

describe("NewLessonInput", () => {
	test("view renders correctly", () => {
		const func = jest.fn()
		const tree = renderer
			.create(
				<NewLessonInput
					name={"test"}
					editable={true}
					selectTextOnFocus={false}
					autoFocus={false}
					setRef={func}
					onFocus={func}
					onBlur={func}
					onChangeText={func}
					iconName="done"
					next={func}
					state={{ initial: "test" }}
					iconType="material"
					onSubmitEditing={func}
					extraPlaceholder={"kiwi"}
					style={{ marginTop: 20 }}
					below={func}
				/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
