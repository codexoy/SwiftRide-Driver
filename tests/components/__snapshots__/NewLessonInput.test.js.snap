// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewLessonInput view renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "rgb(200,200,200)",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "marginLeft": 18,
          "marginRight": 18,
          "marginTop": 20,
          "paddingBottom": 4,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoFocus={false}
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder="[missing \\"he.teacher.new_lesson.test\\" translation]kiwi"
        placeholderTextColor="lightgray"
        rejectResponderTermination={true}
        selectTextOnFocus={false}
        style={
          Object {
            "alignSelf": "center",
            "color": "#000",
            "flex": 1,
            "fontFamily": "Assistant",
            "fontSize": 18,
            "minHeight": 40,
            "paddingLeft": 8,
          }
        }
        testID="testInput"
        textAlign="right"
        underlineColorAndroid="transparent"
        value="value"
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
            underlayColor="white"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 16,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              î¡¶
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "marginTop": 12,
      }
    }
  />,
]
`;
